server/accounts.go

```diff
@@ -4114,18 +4116,73 @@ type transform struct {
 	src, dest string
 	dtoks     []string
 	stoks     []string
-	dtpi      []int8
+	dtpi      [][]int // destination token position indexes
+	dtpinp    []int32 // destination token position index number of partitions
 }
 
-// Helper to pull raw place holder index. Returns -1 if not a place holder.
-func placeHolderIndex(token string) int {
-	if len(token) > 1 && token[0] == '$' {
-		var tp int
-		if n, err := fmt.Sscanf(token, "$%d", &tp); err == nil && n == 1 {
-			return tp
+func getMappingFunctionArgs(functionName string, token string) ([]string, error) {
+	regEx := `{{` + functionName + ` *\((.*)\)}}`
+	r1, err := regexp.Compile(regEx)
+	if err != nil {
+		return nil, err
+	}
+	commandStrings := r1.FindStringSubmatch(token)
+	if len(commandStrings) > 1 {
+		ra, err := regexp.Compile(`,\s*`)
```

Comment:
Same thing here. Make this regex a "constant" by defining it at the top of the file.

Reply:
Optimization done in https://github.com/nats-io/nats-server/pull/2890/commits/ceb47f0298443758546029c469dc29336c515fdb